### PR TITLE
Hotfix 0.37.4

### PR DESCRIPTION
**Pull request by nico-ptrs**

### CODE REVIEW

File: provider-evm/src/main/java/network/arkane/provider/balance/EvmCovalentBalanceStrategy.java [hunk 1]
```diff
@@ -96,38 +77,33 @@ private Balance getNativeBalanceFromCovalent(String account) {
...
-                               .rawBalance(tokenBalance.toString())
-                               .balance(calculateBalance(tokenBalance, tokenInfo.getDecimals()))
-                               .decimals(tokenInfo.getDecimals())
-                               .symbol(tokenInfo.getSymbol())
-                               .name(tokenInfo.getName())
-                               .logo(tokenInfo.getLogo())
-                               .type(tokenInfo.getType())
-                               .transferable(tokenInfo.isTransferable())
-                               .build();
+            return super.getTokenBalances(walletAddress, tokenAddresses);
```
Comment (alpa-coder) [inline provider-evm/src/main/java/network/arkane/provider/balance/EvmCovalentBalanceStrategy.java:83]: maybe add a guard if tokenaddress list  is empty you return them all...?

File: provider-vechain/src/main/java/network/arkane/provider/balance/VechainBalanceGateway.java [hunk 2]
```diff
@@ -55,18 +54,22 @@ public Balance getBalance(final String hexAccount) {
...
 
     @Override
-    public TokenBalance getTokenBalance(final String walletAddress, final String tokenAddress) {
-        final TokenInfo tokenInfo = tokenDiscoveryService.getTokenInfo(SecretType.VECHAIN, tokenAddress).orElseThrow(IllegalArgumentException::new);
-        return getTokenBalance(walletAddress, tokenInfo);
+    public List<TokenBalance> getTokenBalances(final String walletAddress,
+                                               final List<String> tokenAddresses) {
+        final List<TokenInfo> tokenInfos = tokenAddresses.stream()
+                                                         .map(tokenAddress -> tokenDiscoveryService.getTokenInfo(VECHAIN, tokenAddress).orElseThrow(IllegalArgumentException::new))
+                                                         .collect(Collectors.toList());
```
Comment (alpa-coder) [inline provider-vechain/src/main/java/network/arkane/provider/balance/VechainBalanceGateway.java:61]: maybe do this in parallel (also for all the other tokeninfos that are fetched like this)..?

**Review by alpa-coder — CHANGES_REQUESTED** [review]

File: provider-evm/src/main/java/network/arkane/provider/balance/EvmAzraelBalanceStrategy.java [hunk 3]
```diff
@@ -62,34 +57,16 @@ public Balance getBalance(final String account) {
...
-                                                 .name(c.getName())
-                                                 .type("ERC_20")
-                                                 .logo(getLogo(c.getAddress()))
-                                                 .transferable(true)
-                                                 .build())
-                           .orElse(null);
+    @Override
+    public List<TokenBalance> getTokenBalances(final String walletAddress,
+                                               final List<String> tokenAddresses) {
+        final List<String> lowerCaseTokenAddresses = tokenAddresses.stream().map(String::toLowerCase).collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (nico-ptrs) [reply provider-evm/src/main/java/network/arkane/provider/balance/EvmAzraelBalanceStrategy.java:63]: done

File: provider-evm/src/main/java/network/arkane/provider/balance/EvmCovalentBalanceStrategy.java (hunk 1 above)
Comment (nico-ptrs) [reply provider-evm/src/main/java/network/arkane/provider/balance/EvmCovalentBalanceStrategy.java:83]: Done

File: provider-vechain/src/main/java/network/arkane/provider/balance/VechainBalanceGateway.java (hunk 2 above)
Comment (nico-ptrs) [reply provider-vechain/src/main/java/network/arkane/provider/balance/VechainBalanceGateway.java:61]: Done

**Pull request closed without merging**